### ISSUE TITLE
Add missing  @CheckReturnValue

### ISSUE THREAD

**Issue by joel-costigliola** [joel-costigliola/assertj-core#1044]
#### Summary

`@CheckReturnValue` allows to detect invalid AssertJ usage

#### Pull Requests

Two PR needed one for `2.x` branch and one the `master` branch where only Java 8 assertions must be covered (the PR 2.x branch will cover the remaining assertions once merged and 2.x is merged into master).

**Comment by joel-costigliola** [comment]
I'm wondering if we can add the annotation in the parent class/interface only ... that would reduce the amount of work.

**Comment by PascalSchumacher** [comment]
Maybe I miss something, but the new  methods added by the 2.x pull request do not need `@CheckReturnValue`. The annotations added in https://github.com/joel-costigliola/assertj-core/commit/46c04cbb546ff45548d64323bda9be1c4ca3d434 are incorrect, so I removed them in https://github.com/joel-costigliola/assertj-core/commits/2.x

**Comment by joel-costigliola** [comment]
Yeah I realized that when I added `containsAnyOf` for arrays but I haven't yet pushed the commits.
Thanks for fixing this anyway !
I'm also working on this issue, I have found a few places where `@CheckReturnValue` was missing.

**Comment by PascalSchumacher** [comment]
>I have found a few places where @CheckReturnValue was missing.

great :+1:
